### ISSUE TITLE
[Resources: cloudflare_argo] Tiered Cache Topology selection needed

### ISSUE THREAD

**Issue by igloo777** [cloudflare/terraform-provider-cloudflare#996]
### Current Terraform version

Terraform version v0.14.4

### Description

As in https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/argo - cloudflare provider 2.19.2
supports enabling of tiered_caching and smart_routing, But it doesn't support selection of Tiered Cache Topology for tiered_caching.

### Use cases

Currently we have to switch the Tiered Cache Topology in web ui, while the state we keep in terraform. 

### Potential Terraform configuration

```hcl
variable "tiered_caching_topology" {
  type = string
  # Value may be "generic" or "smart"
  default = "generic"
}


resource "cloudflare_argo" "example" {
  zone_id        = "d41d8cd98f00b204e9800998ecf8427e"
  tiered_caching = "on"
  tiered_caching_topology = "smart"
  smart_routing  = "off"
}
```

Currently we have to switch the Tiered Cache Topology in web ui, while the state we keep in terraform. 
The lack of the feature Tiered Cache Topology may be an issue for projects with a large number of domains .

**Comment by jacobbednarz** [comment]
At the time of writing, there isn't a public API endpoint support for this so we can't yet add anything to `cloudflare-go`. Once that lands, we can look at adding support in the Go library before incorporating it here.

**Comment by toddschilling** [comment]
Tracking this issue as Seismic Software has a need for this as well.

**Comment by AlexKasaku** [comment]
This appears to be supported via an undocumented API endpoint.

To enable smart topology:

`PATCH`  `/zones/:zone_id/cache/tiered_cache_smart_topology_enable` with `{"value":"on"}`

To enable smart topology and use generic:

`PATCH`  `/zones/:zone_id/cache/tiered_cache_smart_topology_enable` with `{"value":"off"}`

**Comment by borisceranic** [comment]
@jacobbednarz any chance to have this feature added to cloudflare-go now that there's an (undocumented) API endpoint for this?

I'm hoping that the fact this API is undocumented should not be an issue, because `tiered_caching` was also previously implemented against a similarly undocumented API endpoint 😂

**Comment by jacobbednarz** [comment]
I'm afraid the endpoint needs to be publicly documented before using it in Terraform or the SDK. I'd recommend opening a support ticket to get it routed to the service team to action.

**Comment by evanrappe** [comment]
FYI @jacobbednarz I got this submitted via a support request, and after a few days got this resolution response: 

> Thanks for your continued patience. I will close this ticket as we have filed an internal request to get this resolved.

While it doesn't seem to be done yet, hopefully it will be soon.

**Comment by jacobbednarz** [comment]
this has landed in v3.31.0 via #2101
